fs/fs_zenfs.cc

```diff
@@ -1304,7 +1317,122 @@ void ZenFS::GetZenFSSnapshot(ZenFSSnapshot& snapshot,
     zbd_->GetMetrics()->ReportSnapshot(snapshot);
   }
 
-  zbd_->LogGarbageInfo();
+  if (options.log_garbage_) {
+    zbd_->LogGarbageInfo();
+  }
+}
+
+IOStatus ZenFS::MigrateExtents(
+    const std::vector<ZoneExtentSnapshot*>& extents) {
+  IOStatus s;
+  // Group extents by their filename
+  std::map<std::string, std::vector<ZoneExtentSnapshot*>> file_extents;
+  for (auto* ext : extents) {
+    std::string fname = ext->filename;
+    // We only migrate SST file extents
+    if (ends_with(fname, ".sst")) {
+      file_extents[fname].emplace_back(ext);
+    }
+  }
+
+  for (const auto& it : file_extents) {
+    s = MigrateFileExtents(it.first, it.second);
+    if (!s.ok()) break;
+  }
+  return s;
+}
+
+IOStatus ZenFS::MigrateFileExtents(
+    const std::string& fname,
+    const std::vector<ZoneExtentSnapshot*>& migrate_exts) {
+  IOStatus s = IOStatus::OK();
+  Info(logger_, "MigrateFileExtents, fname: %s, extent count: %lu",
+       fname.data(), migrate_exts.size());
+  // The file may be deleted by other threads, better double check.
+  auto zfile = GetFile(fname);
+  if (zfile == nullptr || zfile->IsOpenForWR()) {
+    return IOStatus::OK();
+  }
+
+  std::vector<ZoneExtent*> new_extent_list;
+  std::vector<ZoneExtent*> extents = zfile->GetExtents();
```

Comment:
This will not copy the extents, right? We'll only get a copy of the vector, so we are still accessing the same extent copy as the readers are using..